### PR TITLE
added range functionality for runes

### PR DESCRIPTION
**Pull request by sealmove**

### CODE REVIEW

**Comment by juancarlospaco** [comment]
Needs changelog entry and more than 1 assert test.
I would make it `func` and `{.inline.}` I dont know.

File: lib/pure/unicode.nim [hunk 1]
```diff
@@ -1256,7 +1258,11 @@ proc isTitle*(s: string): bool {.noSideEffect, procvar, rtl, extern: "nuc$1Str",
     elif rune.isWhiteSpace():
       firstRune = true
 
+proc `..`(`from`, to: Rune): RuneRange =
```
Comment (kaushalmodi) [inline lib/pure/unicode.nim]: These would need export markers too, right?

**Comment by SolitudeSF** [comment]
There is a builtin type for ranges: `HSlice`. Why not use that one instead of introducing new one?

```nim
func contains*(range: HSlice[Rune, Rune], rune: Rune): bool =
  rune >=% range.a and rune <=% range.b
```
this is sufficient for your test to pass.

**Comment by sealmove** [comment]
> There is a builtin type for ranges: `HSlice`. Why not use that one instead of introducing new one?
> 
> ```nim
> func contains*(range: HSlice[Rune, Rune], rune: Rune): bool =
>   rune >=% range.a and rune <=% range.b
> ```
> 
> this is sufficient for your test to pass.

Because HSlice generally works with ordinal types which Rune isn't.
For example `len`, `[]`, `[]=` won't work out of the box, and I don't know how to define them.

**Comment by sealmove** [comment]
Ok... so there is a confict about the `..` operator.
`expression '[toRune("a") .. toRune("z"), toRune("A") .. toRune("Z")]' is of type: array[0..1, RuneRange]`

If Rune is not an ordinal type, and `..` operator can't work, then I think this PR is pointless after all... Maybe it's best if we force the user to simply use the `<%` operator?

**Comment by Araq** [comment]
Unclear benefits so far. Sorry, rejected for now.

**Pull request closed without merging**